assets/javascripts/discourse/components/discourse-post-event/more-menu.gjs

```diff
@@ -0,0 +1,367 @@
+import Component from "@glimmer/component";
+import { fn, hash } from "@ember/helper";
+import EmberObject, { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import DropdownMenu from "discourse/components/dropdown-menu";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { downloadCalendar } from "discourse/lib/download-calendar";
+import { exportEntity } from "discourse/lib/export-csv";
+import { cook } from "discourse/lib/text";
+import i18n from "discourse-common/helpers/i18n";
+import { getAbsoluteURL } from "discourse-common/lib/get-url";
+import I18n from "I18n";
+import DMenu from "float-kit/components/d-menu";
+import { buildParams, replaceRaw } from "../../lib/raw-event-helper";
+import PostEventBuilder from "../modal/post-event-builder";
+import PostEventBulkInvite from "../modal/post-event-bulk-invite";
+import PostEventInviteUserOrGroup from "../modal/post-event-invite-user-or-group";
+
+export default class DiscoursePostEventMoreMenu extends Component {
+  @service appEvents;
+  @service currentUser;
+  @service dialog;
+  @service discoursePostEventApi;
+  @service modal;
+  @service router;
+  @service siteSettings;
+  @service store;
+
+  get expiredOrClosed() {
+    return this.args.event.isExpired || this.args.event.isClosed;
+  }
+
+  get canActOnEvent() {
+    return this.currentUser && this.args.event.canActOnDiscoursePostEvent;
+  }
+
+  get canInvite() {
+    return (
+      !this.expiredOrClosed && this.canActOnEvent && this.args.event.isPublic
+    );
+  }
+
+  get canLeave() {
+    return this.args.event.watchingInvitee && this.args.event.isPublic;
+  }
+
+  get canSeeUpcomingEvents() {
+    return !this.args.event.isClosed && this.args.event.recurrence;
+  }
+
+  get canBulkInvite() {
+    return !this.expiredOrClosed && !this.args.event.isStandalone;
+  }
+
+  @action
+  addToCalendar() {
+    this.menuApi.close();
+
+    const event = this.args.event;
+
+    downloadCalendar(
+      event.name || event.post.topic.title,
+      [
+        {
+          startsAt: event.startsAt,
+          endsAt: event.endsAt,
+        },
+      ],
+      {
+        recurrenceRule: event.recurrenceRule,
+        location: event.url,
+        details: getAbsoluteURL(event.post.url),
+      }
+    );
+  }
+
+  @action
+  sendPMToCreator() {
+    this.menuApi.close();
+
+    this.args.composePrivateMessage(
+      EmberObject.create(this.args.event.creator),
+      EmberObject.create(this.args.event.post)
+    );
+  }
+
+  @action
+  upcomingEvents() {
+    this.router.transitionTo("discourse-post-event-upcoming-events");
+  }
+
+  @action
+  registerMenuApi(api) {
+    this.menuApi = api;
+  }
+
+  @action
+  async inviteUserOrGroup(event) {
```

Comment:
I think all these 7 methods don't need to have an event arg and could use this.args.event directly